ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -865,8 +866,50 @@ let stream_of_channel : Lsp.Cli.Channel.t -> _ = function
     let sockaddr = Unix.ADDR_INET (Unix.inet_addr_loopback, port) in
     socket sockaddr
 
+(* Merlin uses [Sys.command] to run preprocessors and ppxes. We provide an
+   alternative version using the Spawn library for unixes.*)
+let run_in_directory ~prog ~prog_is_quoted:_ ~args ~cwd ?stdin:_ ?stdout
+    ?stderr:_ () =
+  (* Currently we assume that [prog] is always quoted and might contain
+     arguments such as [-as-ppx]. This is due to the way Merlin gets its
+     configuration. Thus we cannot rely on [Filename.quote_command]. *)
+  let args = String.concat ~sep:" " @@ List.map ~f:Filename.quote args in
```

Comment:
Why does we assume that `prog` is always quoted? Shouldn't we inspect `prog_is_quoted` to determine this?

Reply:
Yes, we should have logic to handle both at some point, but as of today these are always "quoted" and the argument is always `true`. I don't think it is worth it to implement the `false` case until Merlin does use it. Since it will require a configuration protocol change it will break the API anyway. Do you prefer that I do test it's value and make an `failwith "Not implemented"` branch ?

Reply:
I think it would have made more sense to just drop the argument then.

Reply:
I agree that it would be cleaner, but we already have 3 released versions of Merlin with it. I think it can wait for the next round of API modification (for example when we will be ready to change how the PPX configuration is obtained). It should not cause any issue while, on the other side, releasing ocaml-lsp now with merlin as a library would provide immediate support for OCaml 5.1 and latest Merlin patches.

Reply:
Okay, let's leave it as a TODO then.